src/main/java/com/hubspot/smtp/client/ResponseHandler.java

```diff
@@ -40,7 +50,13 @@
     // although the future field may have been written in another thread,
     // the compareAndSet call above has volatile semantics and
     // ensures the write will be visible
-    return collector.getFuture();
+    CompletableFuture<List<SmtpResponse>> responseFuture = collector.getFuture();
+
+    // complete exceptionally if the timeout is exceeded
+    Timeout timeout = TIMER.newTimeout(ignored -> responseFuture.completeExceptionally(new TimeoutException()), responseTimeout.toMillis(), TimeUnit.MILLISECONDS);
```

Comment:
note: might want to try/catch the compelteExceptionally since it might throw an ISE